code/modules/shuttles/shuttle_supply_multi.dm

```diff
@@ -0,0 +1,825 @@
+	if(get_movement_direction())
+		moving_status = SHUTTLE_IDLE
+		move_elevator_up_towards(origin, destination)
+	else
+		moving_status = SHUTTLE_IDLE
+		move_elevator_down_towards(origin, destination)
+
+
+///MOVE JANKS vvvvvvvvvvv
+
+/datum/shuttle/ferry/supply/multi/proc/move_elevator_down_from(origin, destination)
+	move(origin, destination)
+	start_gears(SOUTH)
+	if(fake_zlevel)
+		var/animation_timing = 2 SECONDS
+		var/animation_offset = -180
+		if(target_zlevel == fake_zlevel + 1)
+			animation_timing = 7 SECONDS
+			animation_offset = -128
+		elevator_animationz[fake_zlevel].pixel_y = 0
+		//elevator_animationz[fake_zlevel].color = "#000000"
+		playsound(GLOB.supply_elevator_turfs[fake_zlevel], 'sound/machines/asrs_lowering.ogg', 50, 0)
+		animate(elevator_animationz[fake_zlevel], pixel_y = animation_offset, color = "#577eaa", time = animation_timing)
+		spawn(animation_timing)
+			elevator_animationz[fake_zlevel].vis_contents.Cut()
+
+/datum/shuttle/ferry/supply/multi/proc/move_elevator_up_from(origin, destination)
+	move(origin, destination)
+	start_gears(NORTH)
+	if(fake_zlevel)
+		elevator_animationz[fake_zlevel].pixel_y = 0
+		elevator_animationz[fake_zlevel].alpha = 255
+		playsound(GLOB.supply_elevator_turfs[fake_zlevel], 'sound/machines/asrs_raising.ogg', 50, 0)
+		animate(elevator_animationz[fake_zlevel], pixel_y = 96, alpha = 0, time = 2 SECONDS)
+		spawn(2 SECONDS)
+			elevator_animationz[fake_zlevel].vis_contents.Cut()
+		if(target_zlevel == fake_zlevel - 1) //then we need to play unique animation of elevator crawling up @ the target_zlevel
+			elevator_animationz[target_zlevel].pixel_y = -128
+			elevator_animationz[target_zlevel].color = "#577eaa"
+			animate(elevator_animationz[target_zlevel], pixel_y = 0,  color = "#ffffff", time = 7 SECONDS)
+
+// --------------- LEAVING ORIGIN ^^^^^^^ --------- ARRIVING AT DESTINATION vvvvvvvvv
+
+/datum/shuttle/ferry/supply/multi/proc/move_elevator_down_towards(origin, destination)
+	if(target_zlevel)
+		elevator_animationz[target_zlevel].pixel_y = 96
+		elevator_animationz[target_zlevel].alpha = 0
+		animate(elevator_animationz[target_zlevel], pixel_y = 0,  alpha = 255, time = 2 SECONDS)
+
+	spawn(2 SECONDS)
+		move(origin, destination)
+		if(target_zlevel)
+			elevator_animationz[target_zlevel].vis_contents.Cut()
+
+	playsound(locate(Elevator_x,Elevator_y,Elevator_z), 'sound/machines/asrs_lowering.ogg', 50, 0)
+	start_gears(SOUTH)
+
+/datum/shuttle/ferry/supply/multi/proc/move_elevator_up_towards(origin, destination)
+	playsound(locate(Elevator_x,Elevator_y,Elevator_z), 'sound/machines/asrs_raising.ogg', 50, 0)
+
+	if(target_zlevel != fake_zlevel - 1) //if this is true then the animation has already been handled in move_elevator_up_from()
+		elevator_animationz[target_zlevel].pixel_y = -180
+		elevator_animationz[target_zlevel].color = "#577eaa"
+		animate(elevator_animationz[target_zlevel], pixel_y = 0, color = "#ffffff", time = 2 SECONDS)
+
+	spawn(2 SECONDS)
```

Comment:
bestie could you...

for reference it should be something like !!!! pseudocode !!!!
```dm
addtimer(CALLBACK(src, PROC_REF(do_move_up), origin, destination, target_zlevel), 2 SECONDS)

/datum/shuttle/ferry/supply/multi/proc/do_move_up(origin, destination, target_zlevel)
	move(origin, destination)
 	elevator_animationz[target_zlevel].vis_contents.Cut()